### DISCUSSION TITLE
How can I create parent checkbox for subrow

### DISCUSSION THREAD

**Discussion by Jonyberry** [TanStack/table#3975] [Q&A]
Hi,

I have a problem with parent checkbox, I describe it [here](https://github.com/TanStack/table/issues/3965), but apparently it's not a bug. 

When I use expand and select hook, my parent checkbox work as a normal row checkbox, but I need it to work as header checkbox but only for subrows.

Have someone already encountered same issue?

**Comment by samy-tcheik** [comment]
i have same issue anyone can help ?

**Comment by MaherUnlocker** [comment]
i have some issue a*and i send in discord but no response honestly im blocked

**Reply by MaherUnlocker** [reply]
@tannerlinsley  you can guive us some help boss

**Comment by itsmejoeeey** [comment]
I recently opened an issue and PR related to some row-selection issues in #5060. This has now been [fixed in v8.9.10](https://github.com/TanStack/table/releases/tag/v8.9.10).

It's now pretty easy to implement this sort of cascading behavior.

Use the `enableRowSelection` table option to disable selection for the parent rows:
```
const tableInstance = useReactTable({
  columns,
  data,
  defaultColumn,
  ...
  enableRowSelection: (row) => {
    if (row.subRows && row.subRows.length > 0) {
      return false;
    }
    return true;
  },
  ...
});
```

And then render the selection checkboxes in a way similar to:
```
<Checkbox
  checked={row.getIsSelected() || row.getIsAllSubRowsSelected()}
  indeterminate={row.getIsSomeSelected()}
  onClick={() => {
    if (row.getCanExpand()) {
      row.toggleSelected(!row.getIsAllSubRowsSelected());
    } else {
      row.toggleSelected();
    }
  }}
/>
```

Hope this helps :)
